### ISSUE TITLE
[Bug][zos_job_output] - The entire content of the DDNAME is not getting displayed after using the DDNAME filter

### ISSUE THREAD

**Issue by rishinz** [ansible-collections/ibm_zos_core#354]
Hi Team,

Thanks for your help with the DDNAME filtering, (REFERENCE # ISSUE - [IBM zOS Core - zos_job_output] Ability to display only a particular DDNAME for a job or STC instead of all DDNAMES #329)

On further testing of the DDNAME filtering option for a CICS started task to get the entire JESMSGLG content, I am seeing that the output is getting truncated. 

On not using the DDNAME filter, it is observed that the JESMSGLG is getting carried over to the JESJCL DD. 

So, wanted to check if there is any byte (data) LIMIT to the number of lines in a DDNAME that could be printed or is it something else that is truncating the output? @ddimatos 

I have attached the JSON output from my below ansible use case for your reference. Could you please take a look and suggest.

 **# 1. Command to check JESMSGLOG for "CONTROL IS BEING GIVEN""
  tasks:
   - name: JES Job output with ddname
     zos_job_output:
      job_name: "KXDFVU55"
      owner: "KXDCUF"
     register: output**

Thanks for your help as always!

[JSON_OUTPUT.txt](https://github.com/ansible-collections/ibm_zos_core/files/9119517/JSON_OUTPUT.txt)

Cheers!  
Rishin

**Comment by ddimatos** [comment]
@rishinz - thanks for the update, ill need to take a look at the source, I want to say I did fix this in a later version and we may not have backported the truncation. Ill get back to you soon.

**Comment by rishinz** [comment]
Thanks for looking into this @ddimatos

**Comment by ddimatos** [comment]
@rishinz would you be willing to try out a fixtest (unpublished release) first to ensure my fix will work for your case. I would attach it as a tar.gz and provide you the local install command, it would be based on 1.3.5.

**Comment by ddimatos** [comment]
Here is the change to try out, I backported it from 1.4.0-beta.1 to try out. From where the archive is located install it with: `ansible-galaxy collection install -f ibm-ibm_zos_core-*`.

Here is the checksum:
```
$ md5 ibm-ibm_zos_core-1.3.5-patch.tar.gz
   MD5 (ibm-ibm_zos_core-1.3.5-patch.tar.gz) = 6013d1b17afd8d909782c0373cc9775c
```
[ibm-ibm_zos_core-1.3.5-patch.tar.gz](https://github.com/ansible-collections/ibm_zos_core/files/9164949/ibm-ibm_zos_core-1.3.5-patch.tar.gz)

I ran it against a job before and after, before had 133 lines while after the fix had 366 lines, appears to have pulled in all the ddname content. 
File is attached.

**Comment by rishinz** [comment]
Hi Demetri - thanks for the fix. Since our Linux systems are currently undergoing an upgrade, it would take some time for me test this fix. I would keep you posted as soon as I am able to test.

Thanks for your help!

Cheers!

**Comment by ddimatos** [comment]
Thank you for the update @rishinz - we can keep this open for a few more weeks, ill check back. I am interested in knowing if this resolves your issue so that if we  happen to patch 1.3.x again we could include it. We don't currently have any plans for 1.3.x, but we have several new versions coming out through the remainder of the year.

**Comment by vit-tomica** [comment]
Hello @ddimatos, I ran into the original problem as well and found this opened issue. I installed the backported patch and I can confirm that I can now receive the complete joblog output BUT I receive the complete output in each DDNAME of a joblog. In other words, each DDNAME of a joblog contains an identical, complete joblog output. 
Hope this helps.

Thanks, Vitek.

**Comment by ddimatos** [comment]
Hi @vit-tomica - thank you, ill take a look at the output closer, I am pretty sure the backport was correct but only did a quick visual inspection, allow me to look closer. This code that performs these operations rely on SDSF, there are plans to have a low level implementation to bypass that requirement so we should see different behavior then but Ill look at the current code.

**Comment by vit-tomica** [comment]
Thank you @ddimatos. I think the SDSF requirement might be the problem. I use SYSVIEW on my system and it seems the SDSF is not configured on the system.

**Comment by rishinz** [comment]
Hi Demetri - Due to unavailability of the Linux system, I am still unable to test this.

Appreciate your patience in keeping it open and thanks for your help!

Cheers!

**Comment by ddimatos** [comment]
Merged this into release branch 1.3.6 [here](https://github.com/ansible-collections/ibm_zos_core/blob/161ec0373f680900011ff676d7679c7508fee9da/plugins/module_utils/job.py#L336) , branch is going to release process and has passed regression. Will close this issue given that users reported the fix worked, also note I backported this fix from 1.4.0-beta.1 so it will be there going forward.

**Comment by rishijoshi07** [comment]
I am getting the issue where I can see all the DD name's JSON output have same content in JESMSGLG and JESJCL and JESYSMSG on ibm-ibm_zos_core-1.4.0-beta.1 and ibm-ibm_zos_core-1.4.0-beta.2 both also I am getting issue on 1.4.0.

**Comment by ddimatos** [comment]
@rishijoshi07 , thank you for reaching out to us. 

I will need to verify, but I do recall in 1.4.0 that the issue remains since it was corrected in 1.5.0-beta.1 soon to be promoted to 1.5.0. As of right now there is no issue open for `zos_job_output` returning the same content for all the DD's, feel free to open one, this issue  #354 was for content being truncated.

The code base is completely different in 1.5.0, where we use C to obtain the job output and in 1.4 it is REXX, thus there is no option to backport only to correct the REXX and unless a user can not upgrade to 1.5.0 and its dependencies, its not likely a candidate to be corrected in 1.4.x. 

Let us know, and thank you.
